src/payment/is-vaulted-instrument.ts

```diff
@@ -1,5 +1,11 @@
-import { PaymentInstrument, VaultedInstrument } from './payment';
+import { AdyenV2Instrument, FormattedPayload, PaymentInstrument, VaultedInstrument } from './payment';
 
 export default function isVaultedInstrument(instrument: PaymentInstrument): instrument is VaultedInstrument {
+    const formattedPayload = (instrument as FormattedPayload<AdyenV2Instrument>).formattedPayload;
```

Comment:
I don't think this is correct, because this type checker is to determine whether `instrument` is the type of `VaultedInstrument`. It looks like you want to check if `instrument` is the type of `AdyenV2Instrument`. As you can tell, `AdyenV2Instrument` and `VaultedInstrument` are two separate types. Even when you go with the suggestion I've made regarding the ditching the provider-specific interface in favour of a common interface, it will still be different to `VaultedInstrument`. What you should do instead to define a new type checking function.